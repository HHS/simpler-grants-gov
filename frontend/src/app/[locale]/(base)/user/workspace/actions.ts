"use server";

import { getSession } from "src/services/auth/session";
import { inviteUserToOrganization } from "src/services/fetch/fetchers/organizationsFetcher";
import { OrganizationInviteRecord } from "src/types/organizationTypes";
import { z } from "zod";

import { getTranslations } from "next-intl/server";

export type OrganizationInviteValidationErrors = {
  email?: string[];
  role?: string[];
};

export type OrganizationInviteResponse = {
  data?: OrganizationInviteRecord;
  errorMessage?: string;
  validationErrors?: OrganizationInviteValidationErrors;
  success?: boolean;
};

const validateInviteUserAction = async (formData: FormData) => {
  const t = await getTranslations("ManageUsers.inviteUser.validationErrors");
  const schema = z.object({
    email: z.string().min(1, { message: t("email") }),
    role: z.string().min(1, { message: t("role") }),
  });

  const validatedFields = schema.safeParse({
    email: formData.get("email"),
    role: formData.get("role"),
  });

  // Return early if the form data is invalid (server side validation!)
  if (!validatedFields.success) {
    return validatedFields.error.flatten().fieldErrors;
  }
};

export const inviteUserAction = async (
  _prevState: unknown,
  formData: FormData,
  organizationId: string,
): Promise<OrganizationInviteResponse> => {
  const session = await getSession();

  if (!session || !session.token || !session.user_id) {
    return {
      errorMessage: "Not logged in",
    };
  }

  const validationErrors = await validateInviteUserAction(formData);
  if (validationErrors) {
    return {
      validationErrors,
    };
  }

  const requestData = {
    email: formData.get("email") as string,
    roleId: formData.get("role") as string,
    organizationId,
  };

  let inviteUserResponse: OrganizationInviteRecord;
  try {
    inviteUserResponse = await inviteUserToOrganization(
      session.token,
      requestData,
    );
    return { data: inviteUserResponse, success: true };
  } catch (e) {
    const error = e as Error;
    console.error(
      `Error inviting user to org - ${error.message} ${error.cause?.toString() || ""}`,
    );
    return {
      errorMessage: error.message,
    };
  }
};
