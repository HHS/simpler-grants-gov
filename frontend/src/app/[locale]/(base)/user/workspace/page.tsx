import { Metadata } from "next";
import { getSession } from "src/services/auth/session";
import withFeatureFlag from "src/services/featureFlags/withFeatureFlag";
import { getUserOrganizations } from "src/services/fetch/fetchers/organizationsFetcher";
import { getUserDetails } from "src/services/fetch/fetchers/userFetcher";
import { Organization } from "src/types/applicationResponseTypes";
import { LocalizedPageProps } from "src/types/intl";

import { useTranslations } from "next-intl";
import { getTranslations } from "next-intl/server";
import Link from "next/link";
import { redirect } from "next/navigation";
import { ErrorMessage, GridContainer } from "@trussworks/react-uswds";

export async function generateMetadata({
  params,
}: LocalizedPageProps): Promise<Metadata> {
  const { locale } = await params;
  const t = await getTranslations({ locale });
  const meta: Metadata = {
    title: t("UserWorkspace.pageTitle"),
    description: t("Index.metaDescription"),
  };
  return meta;
}

const UserOrganizationsList = ({
  userOrganizations,
}: {
  userOrganizations: Organization[];
}) => {
  const t = useTranslations("UserWorkspace");
  return (
    <>
      <h2>{t("organizations")}</h2>
      <ul>
        {userOrganizations.map((userOrganization) => (
          <li key={userOrganization.organization_id}>
            <Link href={`/organization/${userOrganization.organization_id}`}>
              {userOrganization.sam_gov_entity.legal_business_name}
            </Link>
          </li>
        ))}
      </ul>
    </>
  );
};

async function UserWorkspace() {
  const t = await getTranslations("UserWorkspace");

  const session = await getSession();
  if (!session?.email) {
    // this won't happen, as email is required on sessions, and we're wrapping this in an auth gate in the layout
    console.error("no user session, or user has no email address");
    return;
  }
  let userDetails;
  let userOrganizations;
  const userDetailsPromise = getUserDetails(session.token, session.user_id);
  const userOrganizationsPromise = getUserOrganizations(
    session.token,
    session.user_id,
  );
  try {
    [userDetails, userOrganizations] = await Promise.all([
      userDetailsPromise,
      userOrganizationsPromise,
    ]);
  } catch (e) {
    console.error("Unable to fetch user details", e);
  }

  return (
    <GridContainer className="padding-top-2 tablet:padding-y-6">
      <h1>{t("title")}</h1>
      {userDetails && userOrganizations ? (
        <UserOrganizationsList userOrganizations={userOrganizations} />
      ) : (
        <ErrorMessage>{t("fetchError")}</ErrorMessage>
      )}
    </GridContainer>
  );
}

export default withFeatureFlag<object, never>(
  UserWorkspace,
  "userAdminOff",
  () => redirect("/maintenance"),
);
