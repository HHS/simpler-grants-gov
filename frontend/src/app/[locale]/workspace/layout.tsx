import { LayoutProps } from "src/types/generalTypes";

import BetaAlert from "src/components/BetaAlert";
import { AuthenticationGate } from "src/components/user/AuthenticationGate";

export default function WorkspaceLayout({ children }: LayoutProps) {
  return (
    <AuthenticationGate>
      <BetaAlert containerClasses="margin-bottom-5" />
      {children}
    </AuthenticationGate>
  );
}
