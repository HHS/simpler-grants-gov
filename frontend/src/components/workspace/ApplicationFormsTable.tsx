import { ApplicationFormDetail } from "src/types/applicationResponseTypes";
import { CompetitionForms } from "src/types/competitionsResponseTypes";
import { FormDetail } from "src/types/formResponseTypes";

import { useTranslations } from "next-intl";
import Link from "next/link";
import { Table } from "@trussworks/react-uswds";

import { USWDSIcon } from "src/components/USWDSIcon";

export const selectFormsByRequired = ({
  forms,
  required,
}: {
  forms: CompetitionForms;
  required: boolean;
}): FormDetail[] => {
  return forms.reduce<FormDetail[]>((acc, item) => {
    if (item.is_required === required) {
      acc.push(item.form);
    }
    return acc;
  }, []);
};

const selectApplicationForm = ({
  applicationForms,
  formId,
}: {
  applicationForms: ApplicationFormDetail[];
  formId: string;
}) => {
  return applicationForms
    ? applicationForms.find(
        (applicationForm) => applicationForm.form_id === formId,
      )
    : null;
};

export const ApplicationFormsTable = ({
  applicationForms,
  applicationId,
  forms,
}: {
  applicationForms: ApplicationFormDetail[];
  applicationId: string;
  forms: CompetitionForms;
}) => {
  const requiredForms = selectFormsByRequired({ forms, required: true });
  const conditionalRequiredForms = selectFormsByRequired({
    forms,
    required: false,
  });
  const t = useTranslations("Application.competitionFormTable");

  return (
    <>
      <h3>{t("requiredForms")}</h3>
      <ApplicationTable
        forms={requiredForms}
        applicationForms={applicationForms}
        applicationId={applicationId}
      />
      {conditionalRequiredForms.length > 0 && (
        <>
          <h3>{t("conditionalForms")}</h3>
          <p>{t("conditionalFormsDescription")}</p>
          <ApplicationTable
            forms={conditionalRequiredForms}
            applicationForms={applicationForms}
            applicationId={applicationId}
          />
        </>
      )}
    </>
  );
};

const selectApplicationFormById = ({
  forms,
  formId,
}: {
  forms: FormDetail[];
  formId: string;
}): FormDetail | undefined => {
  return forms.find((form) => form.form_id === formId);
};

const ApplicationTable = ({
  applicationForms,
  applicationId,
  forms,
}: {
  applicationForms: ApplicationFormDetail[];
  applicationId: string;
  forms: FormDetail[];
}) => {
  const t = useTranslations("Application.competitionFormTable");

  return (
    <Table className="width-full overflow-wrap">
      <thead>
        <tr>
          <th scope="col" className="bg-base-lightest padding-y-205">
            {t("status")}
          </th>
          <th scope="col" className="bg-base-lightest padding-y-205">
            {t("form")}
          </th>
          <th scope="col" className="bg-base-lightest padding-y-205">
            {t("instructions")}
          </th>
          <th scope="col" className="bg-base-lightest padding-y-205">
            {t("updated")}
          </th>
        </tr>
      </thead>
      <tbody>
        {applicationForms.map((form, index) => (
          <tr key={index}>
            <td data-label={t("status")}>
              <CompetitionStatus
                applicationForms={applicationForms}
                formId={form.form_id}
              />
            </td>
            <td data-label={t("form")}>
              <FormLink
                formId={form.form_id}
                forms={forms}
                applicationId={applicationId}
                appFormId={form.application_form_id}
              />
            </td>
            <td data-label={t("instructions")}>
              <InstructionsLink
                forms={forms}
                formId={form.form_id}
                text={t("downloadInstructions")}
                unavailableText={t("attachmentUnavailable")}
              />
            </td>
            <td data-label={t("updated")}>
              <div> -- </div>
            </td>
          </tr>
        ))}
      </tbody>
    </Table>
  );
};

const CompetitionStatus = ({
  formId,
  applicationForms,
}: {
  applicationForms: ApplicationFormDetail[];
  formId: string;
}) => {
  const t = useTranslations("Application.competitionFormTable.statuses");
  const applicationForm = selectApplicationForm({ formId, applicationForms });

  if (applicationForm?.application_form_status === "in_progress") {
    return (
      <div className="display-flex flex-align-center text-bold icon-active">
        <USWDSIcon name="loop" className="margin-right-2px" />
        {t("in_progress")}
      </div>
    );
  } else if (applicationForm?.application_form_status === "complete") {
    return (
      <div className="display-flex flex-align-center text-bold">
        <USWDSIcon name="check" className="text-primary margin-right-2px" />
        {t("complete")}
      </div>
    );
  } else {
    return <>-</>;
  }
};

const InstructionsLink = ({
  formId,
  forms,
  text,
  unavailableText,
}: {
  formId: string;
  forms: FormDetail[];
  text: string;
  unavailableText: string;
}) => {
  const instructions = selectApplicationFormById({
    forms,
    formId,
  })?.form_instruction;
  const downloadPath = instructions?.download_path || null;
  return (
    <>
      {downloadPath ? (
        <Link
          className="display-flex flex-align-center font-sans-2xs"
          href={downloadPath}
        >
          <USWDSIcon name="save_alt" className="margin-right-05" />
          {text}
        </Link>
      ) : (
        <>{unavailableText}</>
      )}
    </>
  );
};

const FormLink = ({
  formId,
  forms,
  applicationId,
  appFormId,
}: {
  formId: string;
  forms: FormDetail[];
  applicationId: string;
  appFormId: string;
}) => {
  const formName = selectApplicationFormById({
    forms,
    formId,
  })?.form_name;

  return (
    <>
      {formName && (
        <Link
          className="text-bold"
          href={`/workspace/applications/application/${applicationId}/form/${appFormId}`}
        >
          {formName}
        </Link>
      )}
    </>
  );
};
