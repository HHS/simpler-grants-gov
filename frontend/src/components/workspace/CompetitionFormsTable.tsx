import { CompetitionForms } from "src/types/competitionsResponseTypes";
import { FormDetail, FormInstruction } from "src/types/formResponseTypes";

import { useTranslations } from "next-intl";
import Link from "next/link";
import { Table } from "@trussworks/react-uswds";

import { USWDSIcon } from "src/components/USWDSIcon";

export const CompetitionFormsTable = ({
  forms,
  applicationId,
}: {
  forms: CompetitionForms;
  applicationId: string;
}) => {
  const requiredForms = selectForms({ forms, required: true });
  const conditionalRequiredForms = selectForms({ forms, required: false });
  const t = useTranslations("Application.competitionFormTable");

  return (
    <>
      <h3>{t("requiredForms")}</h3>

      <CompetitionTable forms={requiredForms} applicationId={applicationId} />
      <h3>{t("conditionalForms")}</h3>

      <CompetitionTable
        forms={conditionalRequiredForms}
        applicationId={applicationId}
      />
    </>
  );
};

export const selectForms = ({
  forms,
  required,
}: {
  forms: CompetitionForms;
  required: boolean;
}): FormDetail[] => {
  return forms.reduce<FormDetail[]>((acc, item) => {
    if (item.is_required === required) {
      acc.push(item.form);
    }
    return acc;
  }, []);
};

const CompetitionTable = ({
  forms,
  applicationId,
}: {
  forms: FormDetail[];
  applicationId: string;
}) => {
  const t = useTranslations("Application.competitionFormTable");

  return (
    <Table className="width-full overflow-wrap">
      <thead>
        <tr>
          <th scope="col" className="bg-base-lighter">
            {t("status")}
          </th>
          <th scope="col" className="bg-base-lighter">
            {t("form")}
          </th>
          <th scope="col" className="bg-base-lighter">
            {t("attachment")}
          </th>
          <th scope="col" className="bg-base-lighter">
            {t("instructions")}
          </th>
          <th scope="col" className="bg-base-lighter">
            {t("updated")}
          </th>
        </tr>
      </thead>
      <CompetitionTableBody forms={forms} applicationId={applicationId} />
    </Table>
  );
};

const CompetitionTableBody = ({
  forms,
  applicationId,
}: {
  forms: FormDetail[];
  applicationId: string;
}) => {
  const t = useTranslations("Application.competitionFormTable");

  return (
    <tbody>
      {forms.map((form, index) => (
        <tr key={index}>
          <td data-label={t("status")}>--</td>
          <td data-label={t("form")}>
            <Link
              className="text-bold"
              href={`/workspace/applications/application/${applicationId}/form/${form.form_id}`}
            >
              {form.form_name}
            </Link>
          </td>
          <td data-label={t("attachment")}>
            <div> -- </div>
          </td>
          <td data-label={t("instructions")}>
            <InstructionsLink
              instsruction={form.form_instruction}
              text={t("downloadInstructions")}
            />
          </td>
          <td data-label={t("updated")}>
            <div> -- </div>
          </td>
        </tr>
      ))}
    </tbody>
  );
};

const InstructionsLink = ({
  instsruction,
  text,
}: {
  instsruction: FormInstruction;
  text: string;
}) => {
  return (
    <Link
      className="display-flex flex-align-center font-sans-2xs"
      href={instsruction.download_path}
    >
      <USWDSIcon name="save_alt" className="margin-right-05" />
      {text}
    </Link>
  );
};
