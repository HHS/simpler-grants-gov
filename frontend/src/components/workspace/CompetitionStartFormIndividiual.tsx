"use client";

import { useTranslations } from "next-intl";
import { RefObject } from "react";
import {
  Button,
  ErrorMessage,
  FormGroup,
  Label,
  ModalFooter,
  ModalRef,
  ModalToggleButton,
  TextInput,
} from "@trussworks/react-uswds";

const CompetitionStartFormIndividiual = ({
  opportunityTitle,
  error = "",
  loading = false,
  modalRef,
  onChange,
  onClose,
  onSubmit,
  validationError = "",
}: {
  opportunityTitle: string;
  loading?: boolean;
  error?: string;
  modalRef: RefObject<ModalRef | null>;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  onClose: () => void;
  onSubmit: () => void;
  validationError?: string;
}) => {
  const t = useTranslations("OpportunityListing");

  return (
    <div className="display-flex flex-align-start">
      <FormGroup error={!!validationError} className="margin-top-1">
        <p className="font-sans-sm" data-testid="opportunity-title">
          {opportunityTitle}
        </p>
        <p className="font-sans-3xs">
          {t("startApplicationModal.requiredText")}
        </p>
        <Label
          id={`label-for-name`}
          key={`label-for-name`}
          htmlFor="application-name"
          className="font-sans-2xs"
        >
          {t("startApplicationModal.name")}
          <span>
            <br /> {t("startApplicationModal.description")}
          </span>
        </Label>
        {validationError && <ErrorMessage>{validationError}</ErrorMessage>}
        {error && <ErrorMessage>{error}</ErrorMessage>}

        <TextInput
          type="text"
          name="application-name"
          id="application-name"
          onChange={onChange}
        />
        <ModalFooter>
          <Button
            onClick={onSubmit}
            type="button"
            data-testid="competition-start-individual-save"
            disabled={!!loading}
          >
            {loading ? "loading..." : t("startApplicationModal.saveButtonText")}
          </Button>
          <ModalToggleButton
            modalRef={modalRef}
            closer
            unstyled
            className="padding-105 text-center"
            onClick={onClose}
          >
            {t("startApplicationModal.cancelButtonText")}
          </ModalToggleButton>
        </ModalFooter>
      </FormGroup>
    </div>
  );
};

export default CompetitionStartFormIndividiual;
