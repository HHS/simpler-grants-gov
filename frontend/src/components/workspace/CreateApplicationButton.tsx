"use client";

import { useUser } from "src/services/auth/useUser";
import { startApplication } from "src/services/fetch/fetchers/clientApplicationFetcher";

import { useRouter } from "next/navigation";
import { useCallback, useState } from "react";
import { Button } from "@trussworks/react-uswds";

const CreateApplicationButton = () => {
  const router = useRouter();
  const { user } = useUser();
  const disabled = !user?.token;

  const [loading, setLoading] = useState<boolean>();

  const handleSubmit = useCallback(() => {
    if (!user?.token) return;
    setLoading(true);
    startApplication(user.token)
      .then((data) => {
        const { applicationId } = data;
        router.push(`/formPrototype/${applicationId}`);
      })
      .catch((error) => {
        console.error(error);
      })
      .finally(() => {
        setLoading(false);
      });
  }, [router, user]);
  return (
    <Button disabled={disabled} onClick={handleSubmit} type="submit">
      {loading ? "loading..." : "Create new application"}
    </Button>
  );
};

export default CreateApplicationButton;
