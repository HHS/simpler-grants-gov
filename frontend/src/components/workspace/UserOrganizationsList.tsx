import { Organization } from "src/types/applicationResponseTypes";
import { UserPrivilegesResponse } from "src/types/userTypes";
import { userRoleForOrganization } from "src/utils/userUtils";

import { useTranslations } from "next-intl";
import Link from "next/link";
import { Button, Grid } from "@trussworks/react-uswds";

const OrganizationItem = ({
  organization,
  role,
}: {
  organization: Organization;
  role: string;
}) => {
  const t = useTranslations("UserWorkspace");
  return (
    <li className="border-base-lighter border-1px padding-2 margin-top-2">
      <Grid row>
        <Grid tablet={{ col: 6 }}>
          <div className="font-sans-2xs text-base-dark">{role}</div>
          <h3 className="margin-top-0">
            {organization.sam_gov_entity.legal_business_name}
          </h3>
        </Grid>
        <Grid tablet={{ col: 6 }} className="flex-align-self-end text-right">
          <Link href={`/organization/${organization.organization_id}`}>
            <Button type="button">{t("organizationButtons.view")}</Button>
          </Link>
        </Grid>
      </Grid>
    </li>
  );
};

const NoOrganizations = () => {
  const t = useTranslations("UserWorkspace.noOrganizations");
  return (
    <li className="border-base-lighter border-1px padding-2 margin-top-2">
      <h3>{t("title")}</h3>
      <div>{t("description")}</div>
    </li>
  );
};

export const UserOrganizationsList = ({
  userOrganizations,
  userRoles,
}: {
  userOrganizations: Organization[];
  userRoles: UserPrivilegesResponse;
}) => {
  const t = useTranslations("UserWorkspace");
  return (
    <div className="margin-top-4">
      <h2>{t("organizations")}</h2>
      <ul className="usa-list--unstyled">
        {!userOrganizations.length ? (
          <NoOrganizations />
        ) : (
          <>
            {userOrganizations.map((userOrganization) => (
              <OrganizationItem
                organization={userOrganization}
                role={userRoleForOrganization(userOrganization, userRoles)}
                key={userOrganization.organization_id}
              />
            ))}
          </>
        )}
      </ul>
    </div>
  );
};
