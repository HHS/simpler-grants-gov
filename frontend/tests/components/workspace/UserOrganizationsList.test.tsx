import { render, screen, within } from "@testing-library/react";
import { axe } from "jest-axe";
import {
  fakeOrganizationDetailsResponse,
  fakeUserPrivilegesResponse,
} from "src/utils/testing/fixtures";
import { useTranslationsMock } from "src/utils/testing/intlMocks";

import React from "react";

import { UserOrganizationsList } from "src/components/workspace/UserOrganizationsList";

jest.mock("next-intl", () => ({
  useTranslations: () => useTranslationsMock(),
}));

// Mock the userRoleForOrganization helper so the component renders a predictable role string
jest.mock("src/utils/userUtils", () => ({
  userRoleForOrganization: jest.fn(() => "Admin"),
}));

const makeOrg = (id: string, name: string) => ({
  organization_id: id,
  sam_gov_entity: {
    ...fakeOrganizationDetailsResponse.sam_gov_entity,
    legal_business_name: name,
  },
});

describe("UserOrganizationsList", () => {
  it("should not have accessibility violations", async () => {
    const { container } = render(
      <UserOrganizationsList
        userOrganizations={[fakeOrganizationDetailsResponse]}
        userRoles={fakeUserPrivilegesResponse}
      />,
    );
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it("matches snapshot with multiple organizations", () => {
    const { container } = render(
      <UserOrganizationsList
        userOrganizations={[
          fakeOrganizationDetailsResponse,
          makeOrg("another", "another fake org"),
        ]}
        userRoles={fakeUserPrivilegesResponse}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it("renders the no-organizations state when list is empty", () => {
    render(
      <UserOrganizationsList
        userOrganizations={[]}
        userRoles={fakeUserPrivilegesResponse}
      />,
    );
    const listItems = screen.getAllByRole("listitem");
    // NoOrganizations renders a single list item with a heading and description
    expect(listItems).toHaveLength(1);
    const heading = within(listItems[0]).getByRole("heading");
    expect(heading).toBeInTheDocument();
    expect(listItems[0].textContent).toBeTruthy();
  });

  it("renders an organization item for each provided organization with role, name and link", () => {
    render(
      <UserOrganizationsList
        userOrganizations={[
          fakeOrganizationDetailsResponse,
          makeOrg("another", "another fake org"),
        ]}
        userRoles={fakeUserPrivilegesResponse}
      />,
    );

    const listItems = screen.getAllByRole("listitem");
    expect(listItems).toHaveLength(2);

    // Check name and role for first item
    expect(
      within(listItems[0]).getByText(
        fakeOrganizationDetailsResponse.sam_gov_entity.legal_business_name,
      ),
    ).toBeInTheDocument();
    expect(within(listItems[0]).getByText("Admin")).toBeInTheDocument();

    // Link should point to organization page
    const links = screen.getAllByRole("link");
    expect(links).toHaveLength(2);
    expect(links[0]).toHaveAttribute(
      "href",
      `/organization/${fakeOrganizationDetailsResponse.organization_id}`,
    );
    expect(links[1]).toHaveAttribute("href", "/organization/another");
  });
});
