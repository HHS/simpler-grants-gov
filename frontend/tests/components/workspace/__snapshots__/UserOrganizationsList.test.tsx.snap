// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserOrganizationsList matches snapshot with multiple organizations 1`] = `
<div
  class="margin-top-4"
>
  <h2>
    organizations
  </h2>
  <ul
    class="usa-list--unstyled"
  >
    <li
      class="border-base-lighter border-1px padding-2 margin-top-2"
    >
      <div
        class="grid-row"
        data-testid="grid"
      >
        <div
          class="tablet:grid-col-6"
          data-testid="grid"
        >
          <div
            class="font-sans-2xs text-base-dark"
          >
            Admin
          </div>
          <h3
            class="margin-top-0"
          >
            Completely Legal Organization
          </h3>
        </div>
        <div
          class="tablet:grid-col-6 flex-align-self-end text-right"
          data-testid="grid"
        >
          <a
            href="/organization/great id"
          >
            <button
              class="usa-button"
              data-testid="button"
              type="button"
            >
              organizationButtons.view
            </button>
          </a>
        </div>
      </div>
    </li>
    <li
      class="border-base-lighter border-1px padding-2 margin-top-2"
    >
      <div
        class="grid-row"
        data-testid="grid"
      >
        <div
          class="tablet:grid-col-6"
          data-testid="grid"
        >
          <div
            class="font-sans-2xs text-base-dark"
          >
            Admin
          </div>
          <h3
            class="margin-top-0"
          >
            another fake org
          </h3>
        </div>
        <div
          class="tablet:grid-col-6 flex-align-self-end text-right"
          data-testid="grid"
        >
          <a
            href="/organization/another"
          >
            <button
              class="usa-button"
              data-testid="button"
              type="button"
            >
              organizationButtons.view
            </button>
          </a>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
